src/components/pages/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,107 @@
+import './MainPage.css';
+import ModalCard from '../../ModalCard/ModalCard';
+import CardsSection from '../../CardsSection/CardsSection';
+import ErrorButton from '../../ErrorBoundary/ErrorButton/ErrorButton';
+import Pagination from '../../Pagination/Pagination';
+import Seacrh from '../../Search/Search';
+import { Context } from '../../../context/context';
+import { useSearchParams } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { ICardDescription, ICards } from '../../../types/interfaces';
+import * as constants from '../../../constants/constants';
+
+export default function MainPage() {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const SearchValue: string | null = localStorage.getItem('stringQuery');
+  const initSearchString: string = SearchValue ? SearchValue : '';
+  const [searchString, setSearchString] = useState(initSearchString);
+
+  const SearchPage: string | null = searchParams.get('page');
+  const initSearchPage: string =
+    SearchPage && +SearchPage > 0 ? SearchPage : '1';
+  const [currentPage, setCurrentPage] = useState(+initSearchPage);
+
+  const [cards, setCards] = useState<ICards>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [allPage, setAllPage] = useState(1);
+  const [linkPrevPage, setLinkPrevPage] = useState('');
+  const [linkNextPage, setLinkNextPage] = useState('');
+
+  const [cardDescription, setCardDescription] =
+    useState<ICardDescription | null>(null);
+  const [modalActive, setModalActive] = useState(false);
+  const [isModalLoading, setIsModalLoading] = useState(false);
+
+  const deleteCardStringQuery = () => {
+    setSearchParams({ name: searchString, page: `${currentPage}` });
+  };
+
```

Comment:
вот пример где можно зюзать useCallback, потому что ты потом передаешь эту функцию в ModalCard
```suggestion
  const deleteCardStringQuery = useCallback(() => {
    setSearchParams({ name: searchString, page: `${currentPage}` });
  }, [currentPage, searchString, setSearchParams]);

```